seqr/utils/file_utils.py

```diff
@@ -2,40 +2,68 @@
 import subprocess
 
 from seqr.utils.logging_utils import SeqrLogger
+from seqr.utils.redis_utils import safe_redis_get_json, safe_redis_set_json
 
 logger = SeqrLogger(__name__)
 
+EXPIRATION_TIME_IN_SECONDS = 3600 - 5
+GS_STORAGE_ACCESS_CACHE_KEY = 'gs_storage_access_cache_entry'
 
-def _run_command(command, user=None):
+
+def run_command(command, user=None):
     logger.info('==> {}'.format(command), user)
     return subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
 
 
 def _run_gsutil_command(command, gs_path, gunzip=False, user=None):
     #  Anvil buckets are requester-pays and we bill them to the anvil project
-    project_arg = '-u anvil-datastorage ' if gs_path.startswith('gs://fc-secure') else ''
+    google_project = get_google_project(gs_path)
+    project_arg = '-u {} '.format(google_project) if google_project else ''
     command = 'gsutil {project_arg}{command} {gs_path}'.format(
         project_arg=project_arg, command=command, gs_path=gs_path,
     )
     if gunzip:
         command += " | gunzip -c -q - "
 
-    return _run_command(command, user=user)
+    return run_command(command, user=user)
 
 
-def _is_google_bucket_file_path(file_path):
+def is_google_bucket_file_path(file_path):
     return file_path.startswith("gs://")
 
 
+def get_google_project(gs_path):
```

Comment:
should be a private function since its never used outside the file

Reply:
After I move the get_gs_rest_api_headers() back, it is used by igv_api.py now.